code/modules/mod/modules/modules_antag.dm

```diff
@@ -0,0 +1,440 @@
+	mask_worn_overlay = TRUE
+	use_mod_colors = TRUE
+	/// Whether or not this module removes pressure protection.
+	var/remove_pressure_protection = TRUE
+	/// Speed added to the control unit.
+	var/speed_added = 0.5
+	/// Speed that we actually added.
+	var/actual_speed_added = 0
+	/// Armor values added to the suit parts.
+	var/list/armor_values = list(MELEE = 25, BULLET = 30, LASER = 15, ENERGY = 15)
+	/// List of parts of the suit that are spaceproofed, for giving them back the pressure protection.
+	var/list/spaceproofed = list()
+
+/obj/item/mod/module/armor_booster/on_part_activation()
+	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD) || mod.get_part_from_slot(ITEM_SLOT_MASK) || mod.get_part_from_slot(ITEM_SLOT_EYES)
+	if(istype(head_cover))
+		head_cover.flash_protect = FLASH_PROTECTION_WELDER
+
+/obj/item/mod/module/armor_booster/on_part_deactivation(deleting = FALSE)
+	if(deleting)
+		return
+	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD) || mod.get_part_from_slot(ITEM_SLOT_MASK) || mod.get_part_from_slot(ITEM_SLOT_EYES)
+	if(istype(head_cover))
+		head_cover.flash_protect = initial(head_cover.flash_protect)
+
+/obj/item/mod/module/armor_booster/on_activation()
+	playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	actual_speed_added = max(0, min(mod.slowdown_active, speed_added))
+	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD) || mod.get_part_from_slot(ITEM_SLOT_MASK) || mod.get_part_from_slot(ITEM_SLOT_EYES)
+	if(istype(head_cover))
+		ADD_TRAIT(mod.wearer, TRAIT_HEAD_INJURY_BLOCKED, MOD_TRAIT)
+	var/list/mod_parts = mod.get_parts(all = TRUE)
+	for(var/obj/item/part as anything in mod.get_parts(all = TRUE))
+		part.armor = part.armor.modifyRating(arglist(armor_values))
+		part.slowdown -= speed_added / length(mod_parts)
+		if(!remove_pressure_protection || !isclothing(part))
+			continue
+		var/obj/item/clothing/clothing_part = part
+		if(clothing_part.clothing_flags & STOPSPRESSUREDAMAGE)
+			clothing_part.clothing_flags &= ~STOPSPRESSUREDAMAGE
+			spaceproofed[clothing_part] = TRUE
+	mod.wearer.update_equipment_speed_mods()
+
+/obj/item/mod/module/armor_booster/on_deactivation(display_message = TRUE, deleting = FALSE)
+	if(!deleting)
+		playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD) || mod.get_part_from_slot(ITEM_SLOT_MASK) || mod.get_part_from_slot(ITEM_SLOT_EYES)
+	if(istype(head_cover))
+		REMOVE_TRAIT(mod.wearer, TRAIT_HEAD_INJURY_BLOCKED, MOD_TRAIT)
+	var/list/mod_parts = mod.get_parts(all = TRUE)
+	var/list/removed_armor = armor_values.Copy()
+	for(var/armor_type in removed_armor)
+		removed_armor[armor_type] = -removed_armor[armor_type]
+	for(var/obj/item/part as anything in mod.get_parts(all = TRUE))
+		part.armor = part.armor.modifyRating(arglist(removed_armor))
```

Comment:
> hi if you have any questions i can answer them here

@Fikou haha, might have to take you up on it.

Did you ever have an issue with module deactivation on equip like this? 

Specifically I seem to be running into this list issue with the armor booster. We are still on the list-based armor datums rather than TG with subtyped armor.

![image](https://github.com/user-attachments/assets/d4881e6d-1f25-4ba6-9ff6-f68ece733e8b)

It's the same code as the activation, its just copied to a new list, set to its inverse, then converted from a list to params by the arglist().

Atleast, thats my understanding of it, I'm still learning list manipulation. Do you have any thoughts?

Reply:
Sorry for bothering you Fikou, I figured it out. I don't know how you got it to work in the original Modsuits, I just had to do some extra list magic to get it to work.

Thank you for your time regardless